src/main/java/net/minecraftforge/coremod/api/ASMAPI.java

```diff
@@ -41,18 +54,47 @@ public int toOpcode() {
         }
     }
 
+    /**
+     * Builds a new {@link MethodInsnNode} with the given parameters. The opcode of the method call is determined by the
+     * given {@link MethodType}.
+     *
+     * @param ownerName        The method owner (class)
+     * @param methodName       The method name
+     * @param methodDescriptor The method descriptor
+     * @param type             The type of method call
+     * @return The built method call node
+     */
     public static MethodInsnNode buildMethodCall(final String ownerName, final String methodName, final String methodDescriptor, final MethodType type) {
         return new MethodInsnNode(type.toOpcode(), ownerName, methodName, methodDescriptor, type==MethodType.INTERFACE);
     }
 
+    /**
+     * Maps a method from the given SRG name to the mapped name at deobfuscated runtime.
+     *
+     * @param name The SRG name of the method
+     * @return The mapped name of the method
+     *
+     * @deprecated Forge no longer uses SRG names in production
+     */
+    @Deprecated(forRemoval = true)
     public static String mapMethod(String name) {
         return map(name, INameMappingService.Domain.METHOD);
     }
 
+    /**
+     * Maps a field from the given SRG name to the mapped name at deobfuscated runtime.
+     *
+     * @param name The SRG name of the field
+     * @return The mapped name of the field
+     *
+     * @deprecated Forge no longer uses SRG names in production
+     */
+    @Deprecated(forRemoval = true)
```

Comment:
We need since annotations in forRemovals so we can automate when they need to be removed.

Reply:
How can I accomplish this?

Reply:
Do you want me to put since "5.1" or since "1.21"?